### ISSUE TITLE
Progressiveness and Timeouts

### ISSUE THREAD

**Issue by ecorm** [wamp-proto/wamp-proto#446]
There is no mention in the spec on how timeouts interact with progressive calls and progressive call results.

In the case of _Progressive Call Results_, what are the semantics for the `timeout` option? Possibilities include:
- The duration limit between the initial call and the first result, and between results thereafter.
- The duration limit between the initial call and the final result.

In the case of Progressive Calls, what happens when
- the first progressive call **has** a `timeout` option, but a subsequent one does not? 
- the first progressive call **does not** have a `timeout` option, but a subsequent does?

Do changes in progressive call `timeout` option "clobber" the call's `timeout` property? Should an absent `timeout` option be considered differently than a zero `timeout` option?

---

My personal view is that "timeout" should mean duration limit for the **next** result. The rationale is that it may not be feasible to predict how much time it will take to receive all progressive results. If the goal of timeouts is to not allow calls to remain pending indefinitely while a callee is hung, then a duration limit for the **next** result is sufficient for that purpose.

~~Also, my view is that a different `timeout` option in a subsequent progressive call should replace the current timeout property of the call, where zero means disabling the timeout.~~ 

~~That leaves the case where a subsequent progressive call `timeout` option is absent (or null): does it mean "use the current timeout property" or "disable the timeout"? I think it should mean "use the current timeout property". That way, a caller only needs to specify the `timeout` option for the first progressive call, and it will remain applicable until explicitly changed or disabled.~~

Edit: On second thought, I think it's best if the `timeout` and `receive_progress` options of the initial progressive call remain sticky for the duration of the progressive call, and not be overridden by subsequent progressive calls. The rationale is to keep the implementation simple and robust. For example, if the timeout property changes in the middle of a bunch of progressive calls, the implementation has to worry about cancelling the current timer deadline.

Addendum: For that matter, all options of subsequent progressive calls should be ignored except for the `progress` option. The initial progressive call is what establishes all other options for the entirety of the progressive call transfer.

**Comment by oberstet** [comment]
> Edit: On second thought, I think it's best if the timeout and receive_progress options of the initial progressive call remain sticky for the duration of the progressive call, and not be overridden by subsequent progressive calls. The rationale is to keep the implementation simple and robust. 

this sounds a good approach: +1

**Comment by KSDaemon** [comment]
When I read this I remembered that I was thinking about that at the time of writing, but yeah, somehow it didn't make it in.

And yeah, I was also thinking of sticking to option values from the very first call and do not take into account even changed values on subsequent calls. 

I checked I save an `invocation_policy` option from the first call in Nexus implementation. But do not do anything else with the rest (actually only `timeout`).
